extensions/broker/src/main/java/de/sovity/extension/broker/service/IdsBrokerServiceImpl.java

```diff
@@ -229,17 +242,51 @@ private HashMap<String, Asset> resolveCreatedAssets(ContractDefinitionCreated co
         var contractDefinition = contractDefinitionStore.findById(contractDefinitionId);
 
         var resourceMap = new HashMap<String, Asset>();
-        var assetsFromContractDefinition = getAssetsFromContractDefinition(contractDefinition);
-        for (var asset : assetsFromContractDefinition) {
-            var resourceId = String.format("%s-%s",
-                    contractDefinition.getId(),
-                    asset.getId());
-            resourceMap.put(resourceId, asset);
+        var contractPolicyId = contractDefinition.getContractPolicyId();
+        var accessPolicyId = contractDefinition.getAccessPolicyId();
+
+        // If there is a REFERRING_CONNECTOR policy in the contractPolicy or accessPolicy the resource won't
+        // be sent to the broker.
+        var contractConstraints = getConstraints(policyDefinitionStore.findById(contractPolicyId));
+        var accessConstraints = getConstraints(policyDefinitionStore.findById(accessPolicyId));
+        var hasReferringPolicy = containsReferringConnectorPolicy(contractConstraints) ||
+                containsReferringConnectorPolicy(accessConstraints);
+
+        if (hasReferringPolicy) {
+            monitor.info("Not publishing resource at broker due to possible policy breach.");
+        } else {
+            var assetsFromContractDefinition = getAssetsFromContractDefinition(contractDefinition);
+            for (var asset : assetsFromContractDefinition) {
+                var resourceId = String.format("%s-%s",
+                        contractDefinition.getId(),
+                        asset.getId());
+                resourceMap.put(resourceId, asset);
+            }
         }
 
         return resourceMap;
     }
 
+    private List<AtomicConstraint> getConstraints(PolicyDefinition contractPolicy) {
+        return contractPolicy
+                .getPolicy()
+                .getPermissions()
+                .stream()
+                .map(Rule::getConstraints)
+                .flatMap(List::stream)
+                .map(AtomicConstraint.class::cast)
+                .toList();
+    }
+
+    private boolean containsReferringConnectorPolicy(List<AtomicConstraint> constraints) {
```

Comment:
Method naming (policies freely adjustable via the settings).

Reply:
done